libhypatia/src/segment.rs

```diff
@@ -0,0 +1,25 @@
+/// Call this macro from within each segment to include all the per-segment boilerplate.
+#[macro_export]
+macro_rules! define_segment {
```

Comment:
The name is specific to segments, but used for both tasks and segments (which are different).

Reply:
I've renamed this to __runtime_boilerplate, which is now called by define_segment! and define_task!, which I think are the two types we are trying to represent with several of these crates?

Reply:
Yes, that sounds good. Thanks!